podcasts/PlaybackManager.swift

```diff
@@ -1567,22 +1568,31 @@ class PlaybackManager: ServerPlaybackDelegate {
             #if !os(watchOS)
                 starCommand.setTitle(title: L10n.starEpisodeShort)
             #endif
-            starCommand.removeTarget(self)
+            starCommand.removeTarget(nil)
             starCommand.addTarget { [weak self] _ -> MPRemoteCommandHandlerStatus in
                 guard let strongSelf = self, let episode = strongSelf.currentEpisode() as? Episode else { return .noActionableNowPlayingItem }
-
-                EpisodeManager.setStarred(true, episode: episode, updateSyncStatus: SyncManager.isUserLoggedIn())
+                EpisodeManager.setStarred(!episode.keepEpisode, episode: episode, updateSyncStatus: SyncManager.isUserLoggedIn())
                 return .success
             }
+            if let episode = self.currentEpisode() {
+                starCommand.isActive = episode.keepEpisode
+            } else {
+                starCommand.isActive = false
+            }
             starCommand.isEnabled = true
         } else {
-            markPlayedCommand.removeTarget(self)
+            markPlayedCommand.removeTarget(nil)
             markPlayedCommand.isEnabled = false
 
-            starCommand.removeTarget(self)
+            starCommand.removeTarget(nil)
             starCommand.isEnabled = false
         }
     }
+    
+    @objc private func handleEpisodeStarredChanged() {
+        guard let episode = self.currentEpisode() else {return}
```

Comment:
[Nit] Small formatting nitpicks. 

```suggestion
        guard let episode = currentEpisode() else { return }
```